modules/ding_seo/ding_seo.module

```diff
@@ -226,3 +228,185 @@ function ding_seo_get_person_nodes(array $names) {
     ];
   }, $names);
 }
+
+
+/**
+ * Implements hook_cron().
+ *
+ * Updates the xmlsitemaptable with new links once a day
+ */
+function ding_seo_cron() {
+  $last_run = variable_get('ding_seo_last_run_sitemap', 0);
+
+  if (REQUEST_TIME > $last_run + 86400) {
+    ding_seo_update_sitemap();
+    watchdog('ding_seo', 'Updated sitemap: %time', array('%time' => REQUEST_TIME), WATCHDOG_INFO);
+    variable_set('ding_seo_last_run_sitemap', REQUEST_TIME);
+  }
+}
+
+/**
+ * Implements hook_xmlsitemap_link_info().
+ */
+function ding_seo_xmlsitemap_link_info() {
+  return [
+    'ting_object' => [
+      'label' => t('Ting objects'),
+    ],
+  ];
+}
+
+/**
+ * Updates the xmlsitemaptable with new links.
+ */
+function ding_seo_update_sitemap() {
+  try {
+    ding_seo_check_sitemap_size();
+    $last_check = variable_get('ding_seo_sitemap_last_check', 0);
+    $objects = ding_seo_get_objects($last_check);
+
+    if (count($objects) > 0) {
+      ding_seo_update_sitemap_table($objects);
+      variable_set('xmlsitemap_regenerate_needed', TRUE);
+      variable_set('ding_seo_sitemap_last_check', time());
+    }
+  } catch (Exception $exception) {
+    watchdog_exception('ding_seo', $exception);
+  }
+
+}
+
+/**
+ * Get the new links from last update.
+ */
+function ding_seo_get_objects($last_check ) {
+  $objects = [];
+
+  $query = db_select('ting_object_sitemap', 't')
+    ->condition('t.created', $last_check , '>')
+    ->fields('t', ['tid', 'ting_object_id'])
+    ->execute();
+
+  while ($value = $query->fetchAssoc()) {
+    $tid = $value['tid'];
+    $objects[$tid ] = $value['ting_object_id'];
+  }
+
+  return $objects;
+}
+
+/**
+ * Insert the new links into xmlsitemaptable.
+ */
+function ding_seo_update_sitemap_table($objects) {
+  $maxid = (int) db_query("SELECT max(id) FROM {xmlsitemap} WHERE type = 'ting_object'")->fetchField() ?: 0;
+  $chunks = array_chunk($objects, 1000, TRUE);
+  foreach ($chunks as $chunk) {
+    // We could use xmlsitemap_link_save but it takes only one link at a time. For performance reasons
+    // we update the database table our selves.
+    $query = db_insert('xmlsitemap')->fields(array('id', 'type', 'loc', 'language', 'priority'));
+    foreach ($chunk as $id => $object) {
+      //If our id isn't the largest somthings wrong.
+      if ($id > $maxid) {
+        $query->values(
+          [
+            'id' => $id,
+            'type' => 'ting_object',
+            'loc' => 'ting/object/' . $object,
+            'language' => 'und',
+            'priority' => 0.5,
+          ]
+        );
+      }
+    }
+    $query->execute();
+  }
+}
+
+/**
+ * Checks wether to put the ting_object into a sitemaps.
+ *
+ *  We only want ting_objects from the Library's own collection.
+ *  And we only want the ones the users have viewed as object or collection.
+ */
+function ding_seo_check_object_for_sitemap($ting_object, $view_mode) {
+  if ($view_mode != 'full' && $view_mode != 'collection_list') {
```

Comment:
Nit: Think this check would be better placed in `ding_seo_ding_entity_view()` and then not passing `$view_mode`